README.rst

```diff
@@ -111,3 +112,6 @@ Again, see the included `LICENSE <LICENSE>`_ file for specific legal details.
 
 .. |circleci| image:: https://circleci.com/gh/twisted/twisted.svg?style=svg
 .. _circleci: https://circleci.com/gh/twisted/twisted
+
+.. |mypy| image:: https://github.com/twisted/twisted/workflows/mypy.yaml/badge.svg
```

Comment:
@rodrigc 
The URL should be `https://github.com/twisted/twisted/workflows/mypy/badge.svg`

This can be changed as drive by with the next PR to be merged.

Thanks!